packages/compass-connections/src/connections-manager.ts

```diff
@@ -157,26 +224,38 @@ export class ConnectionsManager extends EventEmitter {
         dataService.addReauthenticationHandler(this.reAuthenticationHandler);
       }
 
-      this.connectionAttempts.delete(connectionInfo.id);
-      this.dataServices.set(connectionInfo.id, dataService);
+      this.connectionAttempts.delete(adjustedConnectionInfoForConnection.id);
+      this.dataServices.set(
+        adjustedConnectionInfoForConnection.id,
+        dataService
+      );
+
       this.updateAndNotifyConnectionStatus(
-        connectionInfo.id,
+        adjustedConnectionInfoForConnection.id,
         ConnectionsManagerEvents.ConnectionAttemptSuccessful,
-        [connectionInfo.id, dataService]
+        [adjustedConnectionInfoForConnection.id, dataService]
       );
+
+      dataService.on?.('connectionInfoSecretsChanged', () => {
+        onDatabaseOIDCSecretsChange(
```

Comment:
The `connectionInfoSecretsChanged` naming was intentionally not referring to OIDC only, OIDC only happens to be the only authentication mechanism where this currently applies, but the concept of "stored secrets need to be updated" itself should be available regardless of the auth mechanism